.github/workflows/test.yml

```diff
@@ -0,0 +1,60 @@
+on:
+  pull_request:
+    branches:
+      - main
+      - master
+  push:
+    branches:
+      - main
+      - master
+  release:
+    types: [published]
+
+env:
+  GOFLAGS: "-mod=vendor"
+  GO111MODULE: on
+
+jobs:
+  test:
+    name: Test and Publish
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.16'
+
+      - name: Setup
+        run: |
+          make tools
+          go mod vendor
+
+      - name: Code Lint
+        run:  make lint
+
+      - name: Code UnitTest
+        run: make test
+        if: ${{ github.event_name == 'release' }}
+
+      # Only on release
+      # From https://goreleaser.com/ci/actions/?h=gpg#signing
+
+      - name: Import GPG key
+        if: ${{ github.event_name == 'release' }}
+        id: import_gpg
+        uses: crazy-max/ghaction-import-gpg@v6
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+
+      - name: Run GoReleaser
+        if: ${{ github.event_name == 'release' }}
+        uses: goreleaser/goreleaser-action@v6
+        with:
+          version: "latest"
+        env:
+          GITHUB_TOKEN: ${{ secrets.TERRAFORM_PROVIDER_APTIBLE_PAT }}
```

Comment:
Why use a token we created instead of [automatic token authentication](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) as described in the [goreleaser action docs](https://github.com/goreleaser/goreleaser-action?tab=readme-ov-file#workflow)?

```yml
permissions:
  contents: write
...
env:
  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
The token is for this user https://github.com/aptible-terraform-provider-aptible
It's here just because I was copying what we did in Travis, and [I _think_ this is used](https://aptible.slack.com/archives/C022BUSK487/p1698165378258539?thread_ts=1698162564.168259&cid=C022BUSK487) for publishing the provider itself? I don't know if `secrets.GITHUB_TOKEN` can do that because I don't know what "that" really is.

We can try but then I also don't know if we can delete that token/user or not, either 🤷

Reply:
Started/continued  a slack thread here: https://aptible.slack.com/archives/C0237PYRF6D/p1720043117927469?thread_ts=1666805577.957439&cid=C0237PYRF6D